processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -758,11 +757,21 @@ public int size()
           switch (holder.getType()) {
             case LONG:
               builder.setValueType(ValueType.LONG);
-              builder.addSerde(new LongGenericColumnPartSerde(holder.longType, BYTE_ORDER));
+              builder.addSerde(
+                  LongGenericColumnPartSerde.legacySerializerBuilder()
```

Comment:
The main difference between the new and legacy builder is: The new builder holds some tmp file backed writer one should **add** every value to. Instead the legacy one directly holds the target data structure like CompressedLongsIndexedSupplier. In the convert case, we had read the structure out from the v8 index, so we just need to use the legacy builder to hold this structure and flush it out.